include/arch/riscv/arch/64/mode/object/structures.bf

```diff
@@ -64,6 +64,15 @@ block asid_pool_cap {
     field_high  capASIDPool     37
 }
 
+#ifdef CONFIG_ALLOW_SBI_CALLS
+block sbi_cap {
+    field capSBIFIDBadge 64
+    field capType 5
+    field capSBIEIDBadge 32
```

Comment:
I'd combine these two into one 64-bit badge value on 64-bit.

Reply:
I think everything becomes simpler if we do that, especially on 64-bits. Maybe I'm wrong and it turns out worse.

You don't have to print out the EID and FID separately when printing out the error. Just say badge value `x` doesn't match provided EID `y` and FID `z`. That would actually be more useful because then people can see that they need to swap the values when setting the badge.